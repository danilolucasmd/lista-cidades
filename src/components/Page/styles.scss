:local(.component) {
  padding-left: 20px;
}