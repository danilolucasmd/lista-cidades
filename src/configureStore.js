import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import app from './reducers';

const configureStore = () =>
  createStore(
    app,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(thunk)
  );

export default configureStore;
