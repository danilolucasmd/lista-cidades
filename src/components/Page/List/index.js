import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchCities } from 'actionCreators/cities';
import { getVisibleCities } from 'reducers';
import ListItem from './ListItem';
import ListControls from './ListControls';

import styles from './styles.css';

class List extends Component {
  PAGE_SIZE = 10;

  constructor(props) {
    super(props);

    this.state = {
      curPage: 1,
      firstItem: 0,
      lastItem: this.PAGE_SIZE,
    };
  }

  componentDidMount() {
    this.props.fetchCities();
  }

  componentWillReceiveProps() {
    this.setState({
      curPage: 1,
      firstItem: 0,
      lastItem: this.PAGE_SIZE,
    });
  }

  handleNextPage = () => {
    this.setState({
      firstItem: this.state.firstItem + this.PAGE_SIZE,
      lastItem: this.state.lastItem + this.PAGE_SIZE,
      curPage: this.state.curPage + 1,
    })
  }

  handlePreviousPage = () => {
    this.setState({
      firstItem: this.state.firstItem - this.PAGE_SIZE,
      lastItem: this.state.lastItem - this.PAGE_SIZE,
      curPage: this.state.curPage - 1,
    })
  }

  render() {
    return (
      <div className={styles.component}>
        <table>
          <thead>
            <tr>
              <th>Estado</th>
              <th>Cidade</th>
              <th></th>
            </tr>
          </thead>
          <tbody>
            {this.props.data.slice(this.state.firstItem, this.state.lastItem).map((item, index) => {
              return (
                <ListItem
                  key={index}
                  data={item}
                />
              );
            })}
          </tbody>
        </table>

        <ListControls
          nextPage={this.handleNextPage}
          prevPage={this.handlePreviousPage}
          curPage={this.state.curPage}
          lastPage={this.props.data.length / this.PAGE_SIZE}
        />
      </div>
    );
  }
}

List.propTypes = {
  data: PropTypes.array.isRequired,
  fetchCities: PropTypes.func,
};

const mapStateToProps = state => ({
  data: state.filters.city || state.filters.state ? getVisibleCities(state.cities, state.filters) : state.cities,
});

export default connect(mapStateToProps, { fetchCities })(List);