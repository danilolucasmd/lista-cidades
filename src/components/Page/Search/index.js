import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import { changeFilters } from 'actionCreators/filters';

import styles from './styles.css';

class Search extends Component {
  constructor(props) {
    super(props);

    this.state = {
      city: '',
      state: '',
    };
  }

  handleSubmit = e => {
    e.preventDefault();

    this.props.changeFilters(this.state);
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  render() {
    return (
      <form className={styles.component} onSubmit={this.handleSubmit}>
        <label>Cidade:</label>
        <input
          id='city'
          name='city'
          value={this.state.city}
          onChange={this.handleChange}
        />

        <label>Estado:</label>
        <input
          id='state'
          name='state'
          value={this.state.state}
          onChange={this.handleChange}
        />

        <button type='submit'>Buscar</button>
      </form >
    );
  }
}

Search.propTypes = {
  changeFilters: PropTypes.func,
};

export default connect(undefined, { changeFilters })(Search);