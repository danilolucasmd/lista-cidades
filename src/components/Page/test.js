import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import Page from './index';
import configureStore from 'configureStore';
import renderer from 'react-test-renderer';

const store = configureStore();

it('renders without crashing', () => {
  const div = document.createElement('div');
  const wrapper = (
    <Provider store={store}>
      <Page />
    </Provider>
  );

  ReactDOM.render(wrapper, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const wrapper = (
    <Provider store={store}>
      <Page />
    </Provider>
  );

  const tree = renderer.create(
    wrapper
  ).toJSON();
  expect(tree).toMatchSnapshot();
});