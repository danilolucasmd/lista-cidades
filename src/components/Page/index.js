import React from 'react';
import Search from './Search';
import List from './List';

import styles from './styles.css';

const Page = () => (
  <div className={styles.component}>
    <Search />
    <List />
  </div>
);

export default Page;