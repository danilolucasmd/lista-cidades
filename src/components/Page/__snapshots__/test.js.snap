// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className={undefined}
>
  <form
    className={undefined}
    onSubmit={[Function]}
  >
    <label>
      Cidade:
    </label>
    <input
      id="city"
      name="city"
      onChange={[Function]}
      value=""
    />
    <label>
      Estado:
    </label>
    <input
      id="state"
      name="state"
      onChange={[Function]}
      value=""
    />
    <button
      type="submit"
    >
      Buscar
    </button>
  </form>
  <div
    className={undefined}
  >
    <table>
      <thead>
        <tr>
          <th>
            Estado
          </th>
          <th>
            Cidade
          </th>
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
    <div
      className={undefined}
    >
      <button
        onClick={[Function]}
      >
        Anterior
      </button>
      <label>
        1
      </label>
      <button
        onClick={[Function]}
      >
        Pr√≥ximo
      </button>
    </div>
  </div>
</div>
`;
